src/guacd/connection.c

```diff
@@ -70,7 +70,7 @@ static int __write_all(int fd, char* buffer, int length) {
     while (length > 0) {
 
         int written = write(fd, buffer, length);
-        if (written < 0)
+        if (written < 0 || errno > 0)
```

Comment:
The contents of `errno` are not defined for a successful call to `write()`. I don't believe the standard allows us to rely on checking solely `errno` to determine whether `write()` failed.

Reply:
As far as I known from a higher level language, [Go wraps `write` call](https://github.com/golang/sys/blob/904bdc257025c7b3f43c19360ad3ab85783fad78/unix/zsyscall_linux_mips.go#L1413) and return error if `errno` is not zero.

Reply:
Unless the standards behind `errno` or `write()` define what `errno` will be after a successful call to `write()`, it is non-portable to make assumptions about that behavior.

Reply:
Yep. That looks good to me.